kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiServer.scala

```diff
@@ -90,9 +92,19 @@ object KyuubiServer extends Logging {
       s" ${Properties.javaVersion}")
     SignalRegister.registerLogger(logger)
     val conf = new KyuubiConf().loadFileDefaults()
-    UserGroupInformation.setConfiguration(KyuubiHadoopUtils.newHadoopConf(conf))
+    hadoopConf = KyuubiHadoopUtils.newHadoopConf(conf)
+    UserGroupInformation.setConfiguration(hadoopConf)
     startServer(conf)
   }
+
+  private[kyuubi] def getHadoopConf(): Configuration = {
+    hadoopConf
+  }
+
+  private[kyuubi] def reloadHadoopConf(): Unit = synchronized {
```

Comment:
Not related to this PR.
Can we also implement `reloadkyuubiConf`?
Because some configurations are actually for the engine, when the engine parameters related to `kyuubi-defaults.conf` are modified or added, the new engine parameters submitted by the server will not take effect, and only after restarting the server will it take effect.

Reply:
kyuubi conf is widely used and you might be re-initialize many components, so it seems more complex.

Reply:
In this pr, we only refresh the hadoop conf for frontend service for proxy user verification.

For HadoopCredentialsManager conf, it is not refreshed.

Reply:
@cxzl25 

Maybe we can introduce api to refresh the conf for KyuubiSessionManager and only take affect for engine.

Reply:
I agree with you. @turboFei 
Reloading `kyuubiConf` rashly may still have side effects.

Configuration refresh is also implemented in HDFS, using the `hdfs dfsadmin -reconfig` command, it will only refresh the configuration items marked refreshable to protect the configuration.

Because `SessionConfAdvisor` can only set session-level parameters, and then in JDBC mode, the cost of server restart is relatively high. 

Maybe we can design how to achieve better dynamic configuration?